env/export.go

```diff
@@ -5,8 +5,22 @@ import (
 	"strings"
 )
 
-var posixExportLineRegex = regexp.MustCompile("\\Adeclare \\-x ([a-zA-Z_]+[a-zA-Z0-9_]*)(=\")?(.+)?\\z")
-var endsWithUnescapedQuoteRegex = regexp.MustCompile("([^\\\\]\"\\z|\\A\"\\z)")
+var (
+	posixExportLineRegex = regexp.MustCompile("\\Adeclare \\-[aAfFgiIlnrtux]+ ([a-zA-Z_]+[a-zA-Z0-9_]*)(=\")?(.+)?\\z")
+	//                                                             ^
+	//                                                             |
+	//                                           all of the available options for declare
+	endsWithUnescapedQuoteRegex = regexp.MustCompile("([^\\\\]\"\\z|\\A\"\\z)")
+
+	// There are a bunch of types of bash variable that we want to ignore, becuase supporting them
+	// is either a pain, or useless, or both.
+	disallowedDeclareOpts = []string{
+		"a", // declare -a - it's an array, and can't be used in the environment anyway
+		"A", // declare -A - it's an associative array, and also can't be used in the environment
+		"n", // declare -n - it's a reference to another variable, and will be a pain to sort out
+		"i", // declare -i - it's an integer, and will break our parser. We theoretically could support this, but it's a pain
```

Comment:
The full set of options accepted by bash `declare` is:
```
      -a	to make NAMEs indexed arrays (if supported)
      -A	to make NAMEs associative arrays (if supported)
      -i	to make NAMEs have the `integer' attribute
      -l	to convert the value of each NAME to lower case on assignment
      -n	make NAME a reference to the variable named by its value
      -r	to make NAMEs readonly
      -t	to make NAMEs have the `trace' attribute
      -u	to convert the value of each NAME to upper case on assignment
      -x	to make NAMEs export
```
(from `help declare`)

So we're not disallowing everything other than `-x`, but everything that haven't disallowed shouldn't have too much impact on our parser, i think

Reply:
👍🏼